install-on-aws/create_vms.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+source variables.sh
+
+key_pair_id=$(aws ec2 import-key-pair --key-name $key_name --public-key-material "fileb://${public_key_path}" --query KeyPairId --output text)
+
+
+# Create a Virtual Private Cloud 
+vpc_id=$(aws ec2 create-vpc --region $region --cidr-block $vpc_cidr_block --tag-specifications "ResourceType=vpc,Tags=[{Key=Project,Value=${project_name}}]" --query Vpc.VpcId --output text)
+echo "VPC ID: $vpc_id"
+
+# Create Subnet 
+subnet_id=$(aws ec2 create-subnet --vpc-id $vpc_id --cidr-block $subnet_cidr_block --tag-specifications "ResourceType=subnet,Tags=[{Key=Project,Value=${project_name}}]" --query Subnet.SubnetId --output text)
+echo "Subnet ID: $subnet_id"
+
+# Now setup internet-gateway and route-table associated with VPC
+igw_id=$(aws ec2 create-internet-gateway --tag-specifications "ResourceType=internet-gateway,Tags=[{Key=Project,Value=${project_name}}]" --query InternetGateway.InternetGatewayId --output text)
+aws ec2 attach-internet-gateway --vpc-id $vpc_id --internet-gateway-id $igw_id
+
+rtb_id=$(aws ec2 create-route-table --vpc-id $vpc_id --tag-specifications "ResourceType=route-table,Tags=[{Key=Project,Value=${project_name}}]" --query RouteTable.RouteTableId --output text)
+aws ec2 create-route --route-table-id $rtb_id --destination-cidr-block 0.0.0.0/0 --gateway-id $igw_id
+aws ec2 associate-route-table --route-table-id $rtb_id --subnet-id $subnet_id
+
+# Create security-group to control inbound/outbound traffic for EC2 instances
+sg_id=$(aws ec2 create-security-group --group-name "neuraldb-enterprise-node-sg" --description "allow ingress for http/s, ssh, and nomad" --vpc-id $vpc_id --tag-specifications "ResourceType=security-group,Tags=[{Key=Project,Value=${project_name}}]"  --query 'GroupId' --output text)
+echo "Security Group ID: $sg_id"
+
+# Allow certain incoming traffic for instances in security group
+aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 22 --cidr 0.0.0.0/0
+aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 80 --cidr 0.0.0.0/0
+aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 443 --cidr 0.0.0.0/0
+aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 4646 --cidr 0.0.0.0/0
+aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 5432 --cidr 0.0.0.0/0
+aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol all --source-group $sg_id
```

Comment:
Currently we open all required ports for all nodes. Should we have more fine-grained control over these?

Reply:
Yes, we should do that. I think we can do that in a different PR though, since this PR is getting quite large now.